packages/components/src/components/input-wth-checkbox/input-with-checkbox.tsx

```diff
@@ -71,22 +96,24 @@ const InputWithCheckbox = ({
 
     const focusInput = () => {
         setTimeout(() => {
-            const el_input = input_wrapper_ref.current.nextSibling.querySelector('input.dc-input-wrapper__input');
-            el_input.focus();
+            const el_input: HTMLElement | null = (
+                input_wrapper_ref.current?.nextSibling as HTMLElement
+            )?.querySelector?.('input.dc-input-wrapper__input');
+            el_input?.focus?.();
         });
     };
 
-    const changeValue = e => {
+    const changeValue = () => {
         const new_is_checked = !is_checked;
         // e.target.checked is not reliable, we have to toggle its previous value
-        onChange({ target: { name: e.target.name, value: new_is_checked } });
+        onChange?.({ target: { name: checkboxName, value: new_is_checked } });
```

Comment:
Just for my understanding what is the reason behind changing this?

Reply:
I see, let's find a type for event instead of changing the code, because we may have an issue by changing this

Reply:
@niloofar-deriv, I think the issue here is that the `event` handler attached to that function is for a `span` element and `span` elements have no `name` `attribute`. That's why @yaswanth-deriv has removed that parameter, because the name of the checkbox is exposed at the global scope using   `checkboxName`. As seen in the screenshot, that's the same value being passed in to the `<Checkbox />` element. I think this is fine. Please look into it and see

![name](https://user-images.githubusercontent.com/104334373/215637496-e5493695-94b2-452f-a97a-849d7c0de4fa.png)

Reply:
Thank u @jim-deriv for your explanations, @yaswanth-deriv for making sure that this doesn't break anything could u please add this change to the QA checklist so later we can ask them to test it :smile:

Reply:
NP :+1: